### ISSUE TITLE
Fixing maintenance branch

### ISSUE THREAD

**Issue by krischer** [obspy/obspy#2130]
877791d0028c87729236f4c610bd4f66ccf10087 caused `master` to be merged into `maintenance_1.1.x` and I guess it has been accidentally pushed.

I attempted to fix this in the `maintenance-fixed` branch here: https://github.com/obspy/obspy/tree/maintenance-fixed

I used a mixture of rebasing and cherry-picking and it seems to be correct as far as I can tell. We have to force push this on the `maintenance_1.1.x` branch before we can continue. Can someone please double check that I found everything? It should include everything in the current `maintenance_1.1.x` minus the merge of `master` and minus https://github.com/obspy/obspy/pull/2120 plus the one commit immediately following. These will have to be redone after it is fixed.

The damage is limited but after we force push to `maintenance_1.1.x` we should protect the branch and disallow direct pushes.

**Don't merge anything into the maintenance branch until this is resolved!**

**Comment by calum-chamberlain** [comment]
Sorry @krischer this is me :disappointed: definitely accidental, just meant to merge maintenance into master not the other way around (all done on github, rather than on my local, so the push happened immediately).

I started going through checking that commits match, but realised comparing with the current maintenance 1.1.x is stupid. Comparing with the commits that were supposed to be merged in #2120 looks good up until 8d17d997c9065a386b4802217b105686dbc10054 which in #2120 has no commit between that and 0ab5a99f9e828af0a39d4cee99544e2627e4fb76, whereas [your fixed branch](https://github.com/obspy/obspy/tree/maintenance-fixed) has quite a few - it looks strange to have had such a big gap in maintenance 1.1.x though?

**Comment by krischer** [comment]
> Sorry @krischer this is me 😞 definitely accidental, just meant to merge maintenance into master not the other way around (all done on github, rather than on my local, so the push happened immediately).

No problem :-) Stuff like this happens. And we discovered it pretty quickly.


> I started going through checking that commits match, but realised comparing with the current maintenance 1.1.x is stupid. Comparing with the commits that were supposed to be merged in #2120 looks good up until 8d17d99 which in #2120 has no commit between that and 0ab5a99, whereas your fixed branch has quite a few - it looks strange to have had such a big gap in maintenance 1.1.x though?

Yea comparing with the current `maintenance_1.1.x` does not work. Some graphical tool makes it a lot easier.  If you are on OSX you can use this: http://gitup.co/ But there are a bunch of others out there.

So my fixed maintenance branch forks off at a33ec31 which is just before the merge commit of `master` into `maintenance`.

I then added the contents of #2117, #2016, #2113, and #2081 with a mixture of cherry-picking and rebasing.

**Comment by megies** [comment]
As pointed out in #2255 these maintenance backup branches on the main repo can be confusing, so I've pushed them to my fork and will delete them on the main repo. I don't think we'll ever need them again, but they don't hurt on my fork..

 - https://github.com/megies/obspy/tree/maintenance-fixed
 - https://github.com/megies/obspy/tree/maintenance_1.1.x_backup
